app/auth/auth_service.ts

```diff
@@ -24,8 +26,11 @@ export class AuthService {
 
   register() {
     if (!capabilities.auth) return;
-    // Set initially preferred group ID from local storage.
-    rpcService.requestContext.groupId = window.localStorage.getItem(SELECTED_GROUP_ID_LOCAL_STORAGE_KEY) || "";
+    // Set initially preferred group ID from cookie.
+    rpcService.requestContext.groupId =
+      this.getCookie(SELECTED_GROUP_ID_COOKIE) ||
+      window.localStorage.getItem(SELECTED_GROUP_ID_LOCAL_STORAGE_KEY) ||
+      "";
```

Comment:
(thought of this after approving)

we might want to proactively populate the cookie after this line, so that if we remove the localStorage fallback later, there's a better chance of the cookie being available.